api/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -185,7 +194,28 @@ private static int convertMicros(long micros, ChronoUnit granularity) {
     }
   }
 
+  public static long convertNanos(long nanos, ChronoUnit granularity) {
+    if (nanos >= 0) {
+      long epochSecond = Math.floorDiv(nanos, NANOS_PER_SECOND);
+      long nanoAdjustment = Math.floorMod(nanos, NANOS_PER_SECOND);
+      return granularity.between(EPOCH, toOffsetDateTime(epochSecond, nanoAdjustment));
+    } else {
+      // add 1 nano to the value to account for the case where there is exactly 1 unit between
+      // the timestamp and epoch because the result will always be decremented.
+      long epochSecond = Math.floorDiv(nanos, NANOS_PER_SECOND);
```

Comment:
the `else` path doesn't seem to be tested

Reply:
done